Add appledoc action

alexmx opened the pull request:
Generate Apple-like source code documentation from specially formatted source code comments.

``` ruby
appledoc(
      project_name: "MyProjectName",
      project_company: "Company Name",
      company_id: "com.company.id",
      input: "MyProjectSources",
      output: "/tmp/appledoc/myProjectDocs",
      options: "--keep-intermediate-files --search-undocumented-doc",
      warnings: "--warn-missing-output-path --warn-missing-company-id",
      exit_threshold: 2 # Exit code threshold below which 0 is returned
    )
```

alexmx committed:
Add appledoc action.

alexmx committed:
Fixed number input types for appledoc command.

alexmx committed:
Update Actions.md

KrauseFx commented on docs/Actions.md:
````diff
@@ -1321,6 +1321,24 @@ end
+
+```ruby
+appledoc(
+      project_name: "MyProjectName",
````
Could you change from 4 to 2 spaces here?

KrauseFx commented on docs/Actions.md:
````diff
@@ -1321,6 +1321,24 @@ end
+      exit_threshold: 2 # Exit code threshold below which 0 is returned
+    )
+```
+Use ``appledoc --help`` to see the list of all command line options.
````
This should only be one `

KrauseFx commented on lib/fastlane/actions/appledoc.rb:
```diff
@@ -0,0 +1,398 @@
+
+      def self.run(params)
+        unless Helper.test?
+          raise "appledoc not installed".red if `which appledoc`.length == 0
```
Maybe add 
```ruby
Helper.log.info "Install using `brew install homebrew/boneyard/appledoc`"
```
before the `raise`

KrauseFx commented on docs/Actions.md:
```diff
@@ -1321,6 +1321,24 @@ end
+      project_company: "Company Name",
+      company_id: "com.company.id",
+      input: "MyProjectSources",
+      output: "/tmp/appledoc/myProjectDocs",
```
Can we remove the optional parameters, like `output`, `options`, `warnings` and `exit_threshold` here? Would it still work, right?

KrauseFx commented:
@alexmx This looks great, I added some comments before it's ready to merge. Do you already use it for your project?

alexmx committed:
Fixed post review comments. Added multiple value support for one param key.

alexmx commented:
@KrauseFx Thanks for review. I've addressed your comments. Also added multiple value support for the same argument key, like `--ignore 'Folder 1' --ignore 'Folder 2'`.

Yes, I'm using it in my current project. My configuration looks like this:
```ruby
appledoc(
      project_name: "NAME",
      project_company: "COMPANY",
      company_id: "COMPANY ID",
      docset_atom_filename: "COMPANY.atom",
      docset_feed_url: "http://example.com",
      docset_package_url: "http://example.com",
      docset_fallback_url: "http://example.com",
      output: "./docset",
      publish_docset: true,
      docset_platform_family: "ios",
      logformat: "xcode",
      options: "--keep-intermediate-files --no-repeat-first-par --search-undocumented-doc",
      warnings: "--no-warn-invalid-crossref",
      input: "SOURCE_FILES",
      ignore: ['Pods', 'ProjectTests']
    )
```

Regarding your comment for Actions.md file, I would like to keep current configuration as it's quite important for people who will want to integrate it to understand how correctly to pass arguments.

alexmx committed:
Fixed test suite description.

xfreebird commented:
Useful thing, thanks @alexmx  ! :+1: 
I use myself ```appledoc``` in projects.

alexmx commented:
@xfreebird Thank you man!

alexmx commented:
@KrauseFx any thoughts about this PR?

KrauseFx commented:
Sorry for the delay - got somehow lost, reviewing it now!

KrauseFx commented:
Amazing work, thank you so much for your great contribution @alexmx!! :+1: Also, great tests and documentation!
